docs/features/type_checking.md

```diff
@@ -0,0 +1,56 @@
+# Type Checking
+
+The Postgres Language Server validates your SQL queries against your actual database schema. As you type, it checks that your tables exist, columns are spelled correctly, and data types match - just like Postgres would when executing the query.
+
+## How it Works
+
+When you write a SQL query, the language server:
+1. Connects to your database
+2. Asks Postgres to validate your query without running it
+3. Shows any errors directly in your editor
+
+Since it uses your actual database, you get the exact same validation that would happen at runtime - but instantly as you type.
+
+## Supported Statements
+
+Since we are using `EXPLAIN`, type checking is only available for DML statements:
+- `SELECT` statements
+- `INSERT` statements
+- `UPDATE` statements
+- `DELETE` statements
+- Common Table Expressions (CTEs)
```

Comment:
```suggestion
Since we are using `EXPLAIN`, type checking is only available for DML statements:  
- `SELECT` statements  
- `INSERT` statements  
- `UPDATE` statements  
- `DELETE` statements  
- Common Table Expressions (CTEs)  
```